src/Instance.ts

```diff
@@ -1,49 +1,56 @@
-import { getDefaults } from './defaults.js';
-import { Lexer } from './Lexer.js';
-import { Parser } from './Parser.js';
-import { Hooks } from './Hooks.js';
-import { Renderer } from './Renderer.js';
-import { Tokenizer } from './Tokenizer.js';
-import { TextRenderer } from './TextRenderer.js';
-import { Slugger } from './Slugger.js';
+import { _getDefaults } from './defaults.js';
+import { _Lexer } from './Lexer.js';
+import { _Parser } from './Parser.js';
+import { _Hooks } from './Hooks.js';
+import { _Renderer } from './Renderer.js';
+import { _Tokenizer } from './Tokenizer.js';
+import { _TextRenderer } from './TextRenderer.js';
+import { _Slugger } from './Slugger.js';
```

Comment:
As I mentioned already in the description, I had to rename most internal symbols i.e. `Tokenizer` to `_Tokenizer` to fix an issue in the generated `.d.ts` file, which would otherwise look like

```
namespace marked {
   var Tokenizer: typeof Tokenizer; // Cyclic reference while the second Tokenizer actually did not mean the variable, but the import within the file. Also aliasing the classes on import did not fix the issue.
} 
```

Do you have any better idea on how to overcome this issue?
You should be able to revert that once you get rid of the `marked` export and it's extension fields.

Reply:
I don't have a solution, but I'll second that I don't like the underscores.